mux/src/tmux_commands.rs

```diff
@@ -59,148 +84,535 @@ impl TmuxDomainState {
+        }
+
+        Ok(())
+    }
+
+    fn sync_window_state(&self, windows: &[WindowItem]) -> anyhow::Result<()> {
+        let current_session = self.tmux_session.lock().unwrap_or(0);
+        let mux = Mux::get();
+
+        self.create_gui_window();
+        let mut gui_window = self.gui_window.lock();
+        let gui_window_id = match gui_window.as_mut() {
+            Some(x) => x,
+            None => {
+                anyhow::bail!("No tmux gui created");
+            }
+        };
+
+        for window in windows.iter() {
+            if window.session_id != current_session {
+                continue;
             }
 
-            let pane_pty = TmuxPty {
-                domain_id: self.domain_id,
-                reader: output_read,
-                cmd_queue: self.cmd_queue.clone(),
-                master_pane: ref_pane,
-            };
-            let writer = WriterWrapper::new(pane_pty.take_writer()?);
-            let mux = Mux::get();
             let size = TerminalSize {
-                rows: pane.pane_height as usize,
-                cols: pane.pane_width as usize,
+                rows: window.window_height as usize,
+                cols: window.window_width as usize,
                 pixel_width: 0,
                 pixel_height: 0,
                 dpi: 0,
             };
 
-            let child = TmuxChild {
-                active_lock: active_lock.clone(),
-            };
+            let tab = Arc::new(Tab::new(&size));
+            tab.set_title(&format!("{}", &window.window_name));
+            mux.add_tab_no_panes(&tab);
 
-            let terminal = wezterm_term::Terminal::new(
-                size,
-                std::sync::Arc::new(config::TermConfig::new()),
-                "WezTerm",
-                config::wezterm_version(),
-                Box::new(writer.clone()),
-            );
+            let _ = self.add_attached_window(window, &tab.tab_id())?;
 
-            let local_pane: Arc<dyn Pane> = Arc::new(LocalPane::new(
-                local_pane_id,
-                terminal,
-                Box::new(child),
-                Box::new(pane_pty),
-                Box::new(writer),
-                self.domain_id,
-                "tmux pane".to_string(),
-            ));
+            let mut split_stack;
+            let mut split_direction;
 
-            let tab = Arc::new(Tab::new(&size));
-            tab.assign_pane(&local_pane);
-
-            self.create_gui_window();
-            let mut gui_window = self.gui_window.lock();
-            let gui_window_id = match gui_window.as_mut() {
-                Some(x) => x,
-                None => {
-                    anyhow::bail!("No tmux gui created");
+            let mut split_pane_index = 1;
+            for l in &window.layout {
+                match l {
+                    TmuxLayout::SinglePane(p) => {
+                        let local_pane = self.create_pane(&p).unwrap();
```

Comment:
bare unwrap here; in general, if you are in a function that returns a `Result` of some kind in wezterm, you should strive to propagate an error rather than producing a panic.  It's especially important when processing the results from a remote protocol because connection or version mismatching weirdness really shouldn't be able to kill wezterm and break its other sessions that could have critical state for the user!